### PR TITLE
Fixes & enhancements for Expense Invites

### PR DESCRIPTION
**Pull request by Betree**
Require https://github.com/opencollective/opencollective-api/pull/8803
Fix for https://opencollective.slack.com/archives/C6JTTA4SK/p1681217531812429 (introduced by https://github.com/opencollective/opencollective-frontend/pull/8740)
Fix https://github.com/opencollective/opencollective/issues/6659

See missing ID in activity.data.payee:

```sql
select *
from "Activities" a
WHERE a."ExpenseId" = 132669
ORDER BY id DESC
```

### CODE REVIEW

File: components/expenses/ExpenseForm.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ export const prepareExpenseForSubmit = expenseData => {
   const isGrant = expenseData.type === expenseTypes.GRANT;
   const payee =
     expenseData.payee?.isNewUser || expenseData.payee?.isInvite
-      ? pick(expenseData.payee, ['name', 'slug', 'email', 'legalName', 'organization', 'newsletterOptIn'])
+      ? pick(expenseData.payee, ['id', 'name', 'slug', 'email', 'legalName', 'organization', 'newsletterOptIn'])
```
Comment (kewitz) [inline components/expenses/ExpenseForm.js:126]: Perhaps we shouldn't use pick here, see next line how this can mean either legacyId or a gQLv2 id.

Comment (Betree) [reply components/expenses/ExpenseForm.js:126]: I did that to stay close to the initial approach: https://github.com/opencollective/opencollective-frontend/pull/8740/files#diff-fb75990d7ea60a101a18c88911bef1c9c8c9c6cdd0455acbe25096cc35ab6f9aL223. Will ush a safer fix

File: components/expenses/ExpenseInviteNotificationBanner.tsx [hunk 2]
```diff
@@ -128,16 +146,18 @@ const ExpenseInviteNotificationBanner = props => {
...
-              <ResendSignInEmailButton createdUser={props.createdUser} />
-            ) : (
-              <ResendDraftInviteButton expense={props.expense} />
-            )}
-            <StyledLink href="/help" buttonStyle="standard" buttonSize="tiny">
-              <FormattedMessage id="error.contactSupport" defaultMessage="Contact support" />
-            </StyledLink>
-          </Box>
+          {canResendEmail && (
+            <Flex mt="10px" flexWrap="wrap" gap="8px">
```
Comment (kewitz) [inline components/expenses/ExpenseInviteNotificationBanner.tsx:150]: Nice! Gap just makes things so easier and cleaner.

File: components/expenses/ExpenseInviteNotificationBanner.tsx [hunk 3]
```diff
@@ -97,9 +114,10 @@ ResendSignInEmailButton.propTypes = {
 };
 
 const ExpenseInviteNotificationBanner = props => {
+  const canResendEmail = Boolean(props.expense.permissions?.canVerifyDraftExpense);
```
Comment (kewitz) [inline components/expenses/ExpenseInviteNotificationBanner.tsx:117]: Related to the note I left in the API review.

File: components/expenses/ExpenseForm.js [hunk 4]
```diff
@@ -112,19 +113,34 @@ const getDefaultExpense = collective => ({
   },
 });
 
+const CREATE_PAYEE_PROFILE_FIELDS = ['name', 'email', 'legalName', 'organization', 'newsletterOptIn'];
```
Comment (kewitz) [inline components/expenses/ExpenseForm.js:116]: Not sure why we dropped `slug` here.

**Review by kewitz — APPROVED** [review]

**Pull request merged**